### PR TITLE
1.21.3

### PR DESCRIPTION
**Pull request by SuperGamerTron**
```
> Task :1.21.3:compileJava FAILED
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\BlocksDataGenerator.java:13: error: cannot find symbol
import net.minecraft.loot.context.LootContextParameterSet;
                                 ^
  symbol:   class LootContextParameterSet
  location: package net.minecraft.loot.context
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\BiomesDataGenerator.java:108: error: cannot find symbol
        Registry<Biome> biomeRegistry = registryManager.get(RegistryKeys.BIOME);
                                                       ^
  symbol:   method get(RegistryKey<Registry<Biome>>)
  location: variable registryManager of type DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\BlockCollisionShapesDataGenerator.java:28: error: cannot find symbol
        Registry<Block> blockRegistry = DGU.getWorld().getRegistryManager().get(RegistryKeys.BLOCK);
                                                                           ^
  symbol:   method get(RegistryKey<Registry<Block>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\BlocksDataGenerator.java:41: error: cannot find symbol
        return DGU.getWorld().getRegistryManager().get(RegistryKeys.ITEM).stream()
                                                  ^
  symbol:   method get(RegistryKey<Registry<Item>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\BlocksDataGenerator.java:51: error: package LootContextParameterSet does not exist
            LootContextParameterSet.Builder lootContextParameterSet = new LootContextParameterSet.Builder(serverWorld)
                                   ^
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\BlocksDataGenerator.java:51: error: package LootContextParameterSet does not exist
            LootContextParameterSet.Builder lootContextParameterSet = new LootContextParameterSet.Builder(serverWorld)
                                                                                                 ^
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\BlocksDataGenerator.java:148: error: method getOpacity in class AbstractBlockState cannot be applied to given types;
        blockDesc.addProperty("filterLight", defaultState.getOpacity(EmptyBlockView.INSTANCE, BlockPos.ORIGIN));
                                                         ^
  required: no arguments
  found:    EmptyBlockView,BlockPos
  reason: actual and formal argument lists differ in length
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\MaterialsDataGenerator.java:84: error: cannot find symbol
        Registry<Item> itemRegistry = DGU.getWorld().getRegistryManager().get(RegistryKeys.ITEM);
                                                                         ^
  symbol:   method get(RegistryKey<Registry<Item>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\MaterialsDataGenerator.java:114: error: cannot find symbol
        Registry<Item> itemRegistry = DGU.getWorld().getRegistryManager().get(RegistryKeys.ITEM);
                                                                         ^
  symbol:   method get(RegistryKey<Registry<Item>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\EffectsDataGenerator.java:42: error: cannot find symbol
        Registry<StatusEffect> statusEffectRegistry = DGU.getWorld().getRegistryManager().get(RegistryKeys.STATUS_EFFECT);
                                                                                         ^
  symbol:   method get(RegistryKey<Registry<StatusEffect>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\EnchantmentsDataGenerator.java:28: error: cannot find symbol
                .get(RegistryKeys.ENCHANTMENT)
                ^
  symbol:   method get(RegistryKey<Registry<Enchantment>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\EnchantmentsDataGenerator.java:103: error: cannot find symbol
        Registry<Enchantment> enchantmentRegistry = DGU.getWorld().getRegistryManager().get(RegistryKeys.ENCHANTMENT);
                                                                                       ^
  symbol:   method get(RegistryKey<Registry<Enchantment>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\EntitiesDataGenerator.java:39: error: no suitable method found for create(ServerWorld)
            Entity entityObject = entityType.create(minecraftServer.getOverworld());
                                            ^
    method EntityType.create(ServerWorld,Consumer<CAP#1>,BlockPos,SpawnReason,boolean,boolean) is not applicable
      (actual and formal argument lists differ in length)
    method EntityType.create(World,SpawnReason) is not applicable
      (actual and formal argument lists differ in length)
  where CAP#1 is a fresh type-variable:
    CAP#1 extends Entity from capture of ?
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\EntitiesDataGenerator.java:54: error: no suitable method found for create(World)
        Entity entity = entityType.create(DGU.getWorld());
                                  ^
    method EntityType.create(ServerWorld,Consumer<CAP#1>,BlockPos,SpawnReason,boolean,boolean) is not applicable
      (actual and formal argument lists differ in length)
    method EntityType.create(World,SpawnReason) is not applicable
      (actual and formal argument lists differ in length)
  where CAP#1 is a fresh type-variable:
    CAP#1 extends Entity from capture of ?
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\EntitiesDataGenerator.java:115: error: cannot find symbol
        Registry<EntityType<?>> entityTypeRegistry = DGU.getWorld().getRegistryManager().get(RegistryKeys.ENTITY_TYPE);
                                                                                        ^
  symbol:   method get(RegistryKey<Registry<EntityType<?>>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\FoodsDataGenerator.java:46: error: cannot find symbol
        Registry<Item> itemRegistry = DGU.getWorld().getRegistryManager().get(RegistryKeys.ITEM);
                                                                         ^
  symbol:   method get(RegistryKey<Registry<Item>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\ItemsDataGenerator.java:25: error: cannot find symbol
                .filter(otherItem -> sourceItem.canRepair(sourceItemStack, otherItem.getDefaultStack()))
                                               ^
  symbol:   method canRepair(ItemStack,ItemStack)
  location: variable sourceItem of type Item
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\ItemsDataGenerator.java:42: error: cannot find symbol
        DGU.getWorld().getRegistryManager().get(RegistryKeys.ENCHANTMENT).stream()
                                           ^
  symbol:   method get(RegistryKey<Registry<Enchantment>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\ItemsDataGenerator.java:79: error: cannot find symbol
        Registry<Item> itemRegistry = DGU.getWorld().getRegistryManager().get(RegistryKeys.ITEM);
                                                                         ^
  symbol:   method get(RegistryKey<Registry<Item>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\ParticlesDataGenerator.java:29: error: cannot find symbol
        Registry<ParticleType<?>> particleTypeRegistry = DGU.getWorld().getRegistryManager().get(RegistryKeys.PARTICLE_TYPE);
                                                                                            ^
  symbol:   method get(RegistryKey<Registry<ParticleType<?>>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:21: error: cannot find symbol
        return DGU.getWorld().getRegistryManager().get(RegistryKeys.ITEM).getRawId(item);
                                                  ^
  symbol:   method get(RegistryKey<Registry<Item>>)
  location: interface DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:34: error: cannot find symbol
        for (RecipeEntry<?> recipeE : Objects.requireNonNull(DGU.getWorld()).getRecipeManager().values()) {
                                                                                               ^
  symbol:   method values()
  location: interface RecipeManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:45: error: cannot find symbol
                    var matching = stacks.getMatchingStacks();
                                         ^
  symbol:   method getMatchingStacks()
  location: variable stacks of type Optional<Ingredient>
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:70: error: cannot find symbol
                resultObject.addProperty("id", getRawIdFor(sr.getResult(registryManager).getItem()));
                                                             ^
  symbol:   method getResult(DynamicRegistryManager)
  location: variable sr of type ShapedRecipe
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:71: error: cannot find symbol
                resultObject.addProperty("count", sr.getResult(registryManager).getCount());
                                                    ^
  symbol:   method getResult(DynamicRegistryManager)
  location: variable sr of type ShapedRecipe
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:74: error: cannot find symbol
                String id = ((Integer) getRawIdFor(sr.getResult(registryManager).getItem())).toString();
                                                     ^
  symbol:   method getResult(DynamicRegistryManager)
  location: variable sr of type ShapedRecipe
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:107: error: cannot find symbol
                for (Ingredient ingredient : sl.getIngredients()) {
                                               ^
  symbol:   method getIngredients()
  location: variable sl of type ShapelessRecipe
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:108: error: cannot find symbol
                    if (ingredient.isEmpty()) continue;
                                  ^
  symbol:   method isEmpty()
  location: variable ingredient of type Ingredient
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:109: error: cannot find symbol
                    ingredients.add(getRawIdFor(ingredient.getMatchingStacks()[0].getItem()));
                                                          ^
  symbol:   method getMatchingStacks()
  location: variable ingredient of type Ingredient
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:114: error: cannot find symbol
                resultObject.addProperty("id", getRawIdFor(sl.getResult(registryManager).getItem()));
                                                             ^
  symbol:   method getResult(DynamicRegistryManager)
  location: variable sl of type ShapelessRecipe
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:115: error: cannot find symbol
                resultObject.addProperty("count", sl.getResult(registryManager).getCount());
                                                    ^
  symbol:   method getResult(DynamicRegistryManager)
  location: variable sl of type ShapelessRecipe
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\RecipeDataGenerator.java:117: error: cannot find symbol
                recipes.put(getRawIdFor(sl.getResult(registryManager).getItem()), rootRecipeObject);
                                          ^
  symbol:   method getResult(DynamicRegistryManager)
  location: variable sl of type ShapelessRecipe
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\TintsDataGenerator.java:134: error: cannot find symbol
        Registry<Biome> biomeRegistry = registryManager.get(RegistryKeys.BIOME);
                                                       ^
  symbol:   method get(RegistryKey<Registry<Biome>>)
  location: variable registryManager of type DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\generators\TintsDataGenerator.java:135: error: cannot find symbol
        Registry<Block> blockRegistry = registryManager.get(RegistryKeys.BLOCK);
                                                       ^
  symbol:   method get(RegistryKey<Registry<Block>>)
  location: variable registryManager of type DynamicRegistryManager
D:\Desktop\Programs\PrismarineJS\minecraft-data-generator-server\1.21.3\src\main\java\dev\u9g\minecraftdatagenerator\util\EmptyRenderBlockView.java:65: error: cannot find symbol
        return type == LightType.SKY ? getMaxLightLevel() : 0;
                                       ^
  symbol:   method getMaxLightLevel()
  location: class EmptyRenderBlockView
35 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':1.21.3:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --info option to get more log output.
> Run with --scan to get full insights.

BUILD FAILED in 2s
8 actionable tasks: 2 executed, 6 up-to-date

```

### CODE REVIEW

**Comment by rom1504** [comment]
This seems to run properly 

Want to take the output and do a Minecraft data PR so we see if it's good ?

File: settings.gradle [hunk 1]
```diff
@@ -31,5 +31,10 @@ include "common"
         "1.20",
         "1.20.4",
         "1.20.5",
-        "1.21"
+
+        "1.21",
+        "1.21.3"
+
+        
+
```
Comment (AlexProgrammerDE) [inline settings.gradle:39]: Please fix formatting

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -30,7 +30,11 @@ jobs:
           "1.20",
           "1.20.4",
           "1.20.5",
-          "1.21"
+          "1.21",
+
+          "1.21.3"
+
+
```
Comment (AlexProgrammerDE) [inline .github/workflows/build.yml:37]: Please fix formatting

File: 1.21.3/src/main/java/dev/u9g/minecraftdatagenerator/generators/BiomesDataGenerator.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    private static String guessCategoryBasedOnName(String name, String dimension) {
+        if (dimension.equals("nether")) {
+            return "nether";
+        } else if (dimension.equals("end")) {
+            return "the_end";
+        }
+
+        if (name.contains("end")) {
+            System.out.println();
+        }
```
Comment (AlexProgrammerDE) [inline 1.21.3/src/main/java/dev/u9g/minecraftdatagenerator/generators/BiomesDataGenerator.java:34]: You sure this should be left in? looks like debug

**Review by AlexProgrammerDE — CHANGES_REQUESTED** [review]

File: 1.21.3/src/main/java/dev/u9g/minecraftdatagenerator/generators/BiomesDataGenerator.java (hunk 3 above)
Comment (rom1504) [reply 1.21.3/src/main/java/dev/u9g/minecraftdatagenerator/generators/BiomesDataGenerator.java:34]: Gave you contributor access @AlexProgrammerDE so you can fix that kind of stuff directly

**Comment by rom1504** [comment]
Ci fails

**Comment by rom1504** [comment]
@GroobleDierne do you want to try and fix the recipe generator here ?

**Comment by GroobleDierne** [comment]
Will try

**Comment by GroobleDierne** [comment]
I think tints data are incorrect, that's what I get when diffing against 1.21.1 data
image

**Comment by AlexProgrammerDE** [comment]
@rom1504 PR can be merged

**Pull request merged**